index.d.ts

```diff
@@ -0,0 +1,3 @@
+type AsyncFunction<T> = (...args: any[]) => Promise<T>;
```

Comment:
Maybe

```ts
type AsyncFunction<T, A> = (...args: A) => Promise<T>
```

Reply:
Using `(...args: A)` is incorrect, because `A rest parameter must be of an array type`. You could write `(...args: A[])`, but this would lead nowhere since not all arguments have to be of the same type.

Additionally, it would break type inference and TypeScript would refuse to compile in `strict` mode. 

Try [TypeScript playground example](https://www.typescriptlang.org/play/#src=type%20AsyncFunction%3CT%2C%20A%3E%20%3D%20(...args%3A%20A%5B%5D)%20%3D%3E%20Promise%3CT%3E%0D%0A%0D%0Adeclare%20function%20greenlet%3CT%20extends%20AsyncFunction%3CU%2C%20V%3E%2C%20U%2C%20V%3E(fn%3A%20T)%3A%20T%3B%0D%0A%0D%0A%0D%0Aasync%20function%20add(a%3A%20number%2C%20b%3A%20number)%20%7B%0D%0A%20%20return%20a%20%2B%20b%3B%0D%0A%7D%0D%0A%0D%0Aconst%20result%20%3D%20greenlet(add)%3B) with `--strictFunctionTypes`  enabled.